webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { push } from 'connected-react-router';
+import { useHistory } from 'react-router-dom';
+import { DropdownItem, Spinner } from '@patternfly/react-core';
+import { ChartDonut, ChartLegend, ChartLabel } from '@patternfly/react-charts';
+import { STATUS } from 'foremanReact/constants';
+import { get } from 'foremanReact/redux/API';
+import {
+  selectAPIResponse,
+  selectAPIStatus,
+  selectAPIErrorMessage,
+} from 'foremanReact/redux/API/APISelectors';
+import CardTemplate from 'foremanReact/components/HostDetails/Templates/CardItem/CardTemplate';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { insightsCloudUrl } from '../InsightsCloudSync/InsightsCloudSyncHelpers';
+import ErrorState from '../ForemanInventoryUpload/Components/AccountList/Components/ErrorState/ErrorState';
+
+const InsightsTotalRiskCard = ({ hostDetails: { id } }) => {
+  const initialRisks = {
+    1: { value: 0, title: __('Low') },
+    2: { value: 0, title: __('Moderate') },
+    3: { value: 0, title: __('Important') },
+    4: { value: 0, title: __('Critical') },
+    total: 0,
+  };
+  const [totalRisks, setTotalRisks] = useState(initialRisks);
+  const hashHistory = useHistory();
+  const dispatch = useDispatch();
+  const API_KEY = `HOST_${id}_RECOMMENDATIONS`;
+  const status = useSelector(state => selectAPIStatus(state, API_KEY));
+  const error = useSelector(state => selectAPIErrorMessage(state, API_KEY));
+  const { hits = [] } = useSelector(state => selectAPIResponse(state, API_KEY));
+
+  useEffect(() => {
+    id &&
+      dispatch(
+        get({
+          key: API_KEY,
+          url: insightsCloudUrl(`hits/${id}`),
+        })
+      );
+  }, [API_KEY, dispatch, id]);
+
+  // eslint-disable-next-line consistent-return
+  useEffect(() => {
+    if (!hits || !hits.length) return setTotalRisks(initialRisks);
+    const risks = { ...initialRisks };
+    // eslint-disable-next-line camelcase
+    hits.forEach(({ total_risk }) => {
+      risks[total_risk].value += 1;
+    });
+    risks.total = hits.length;
+    setTotalRisks(risks);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [hits.length]);
```

Comment:
In general, using the length as a dependency might create unexpected bugs, i.e the array can be at the same size but with different values, that's why eslint's `exhaustive-deps`  is angry :)

Instead, we can use useSelector's `shallowEqual`
```js
import { useSelector, shallowEqual } from 'react-redux';

const { hits = [] } = useSelector(state => selectAPIResponse(state, API_KEY), shallowEqual);
```